extension/android/src/main/java/org/pytorch/executorch/EValue.java

```diff
@@ -287,4 +289,75 @@ private void preconditionType(int typeCodeExpected, int typeCode) {
   private String getTypeName(int typeCode) {
     return typeCode >= 0 && typeCode < TYPE_NAMES.length ? TYPE_NAMES[typeCode] : "Unknown";
   }
+
+  /**
+   * Serializes an {@code EValue} into a byte array.
+   *
+   * @return The serialized byte array.
+   * @apiNote This method is experimental and subject to change without notice. This does NOT
+   *     supoprt list type.
+   */
+  public byte[] toByteArray() {
+    if (isNone()) {
+      return ByteBuffer.allocate(1).put((byte) TYPE_CODE_NONE).array();
+    } else if (isTensor()) {
+      Tensor t = toTensor();
+      byte[] tByteArray = t.toByteArray();
+      return ByteBuffer.allocate(1 + tByteArray.length)
+          .put((byte) TYPE_CODE_TENSOR)
+          .put(tByteArray)
+          .array();
+    } else if (isBool()) {
+      return ByteBuffer.allocate(2)
+          .put((byte) TYPE_CODE_BOOL)
+          .put((byte) (toBool() ? 1 : 0))
+          .array();
+    } else if (isInt()) {
+      return ByteBuffer.allocate(9).put((byte) TYPE_CODE_INT).putLong(toInt()).array();
+    } else if (isDouble()) {
+      return ByteBuffer.allocate(9).put((byte) TYPE_CODE_DOUBLE).putDouble(toDouble()).array();
+    } else if (isString()) {
+      return ByteBuffer.allocate(1 + toString().length())
+          .put((byte) TYPE_CODE_STRING)
+          .put(toString().getBytes())
+          .array();
+    } else {
```

Comment:
will we add list support as well?

Reply:
Per Jacob, list is an internal dtype within ET runtime. Maybe we should totally get rid of list in java layer. I could double check with the team